src/Directory.java

```diff
@@ -0,0 +1,63 @@
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class Directory implements Item {
+    private String name;
+    private Map<String, Item> contents;  // Can hold both files and directories
+    private Directory parent;
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
+
+    public Directory(String name, Directory parent) {
+        this.name = name;
+        this.contents = new HashMap<>();
+        this.parent = parent;
+        this.createdAt = LocalDateTime.now();
+        this.updatedAt = LocalDateTime.now();
+    }
+
+    @Override
+    public String getName() {
```

Comment:
There is some library in java jisme aap bas ye @getter and @setter kar doge toh vo aa jayega. Better to use that

Reply:
Okay looking into that